rfcs/min-relations.md

```diff
@@ -0,0 +1,273 @@
+    query.set_relation_filters(
+        RelationFilters::new()
+            .add_target_filter::<FriendsWith, _>(dorothy.entity)
+    );
+
+    for source_entity in query.iter(){
+        println!("{} is friends with Dorothy!", source_entity);
+    }
+
+    // RelationFilters are reset at the end of the system, make sure to set them every time your system runs!
+    // You can also reset this manually by applying a new `RelationFilters` to the query
+    // They override existing relation filters, rather than adding to them
+}
+
+// Or even look for some combination of targets!
+fn caught_in_the_middle(mut query: Query<&mut Stress, With<Relation<FriendsWith>>>,
+    dorothy: Res<Dorothy>,
+    elphaba: Res<Elphaba>){
+    
+    // You can directly chain this as set_relation_filters returns &mut Query
+    query.set_relation_filters(
+        // Note that we can set relation filters even if the relation is in a query filter
+        RelationFilters::new()
+            // Note: add_target_filter is additive- i.e. entities this query
+            // matches must be friends with *both* dorothy AND elphaba
+            .add_target_filter::<FriendsWith, _>(dorothy.entity)
+            .add_target_filter::<FriendsWith, _>(elphaba.entity)
+    ).for_each(|mut stress| {
+        println!("{} is friends with Dorothy and Elphaba!", source_entity);
+        stress.val = 1000; 
+    })
+}
+
+/// If we want to combine targets in an "or" fashion, we need a slightly different syntax
+TODO: write example
+
+// Query filters work too!
+fn not_alone(
+    commands: mut Commands, 
+    query: Query<Entity, Without<Relation<FriendsWith>>>,
+    puppy: Res<Puppy>
+) {
+    for lonely_entity in query.iter(){
+        commands.insert_relation(FriendsWith, puppy.entity);
+    }
+}
+
+// So does change detection with Added and Changed!
+fn new_friends(query: Query<&mut Excitement, Added<Relation<FriendsWith>>>){
+    query.for_each(|mut excitement|{excitement.value += 10});
+}
+
+```
+
+TODO: mention relation cleanup on despawn.
+
+Sometimes, we may instead want to look for "entities that are targets of some relation".
+We can do so by iterating over all entities with that relation, and then filtering by target.
+
+```rust
+fn all_children(
+    player_query: Query<Entity, With<Player>>,
+    child_query: Query<&Name, (With<Relation<ChildOf>, Without<Player>)>>,
+) {
+    let player_entity = player_query.single.unwrap();
+    
+    child_query.set_relation_filters(
+        RelationFilters::new()
+            .add_target_filter::<ChildOf, _>(player_entity)
+    );
+
+    for name in child_query.iter() {
+        println!("{} is one of my children.", name) 			
+    }
+}
+```
+
+Finally, you can use the `Entity` returned by your relations to fetch data from the target's components by combining it with `query::get()` or `query::get_component<C>()`.
+
+```rust
+// We need to use a marker component to avoid having multiple mutable references
+fn debts_come_due(
```

Comment:
Replace with:
fn debts_come_due(
    mut debt_query: Query<Relation<Owes>>, 
    mut money_query: Query<&mut Money>,
) {}

Courtesy of @TheRawMeatball.